TempMail/utils/Email.swift

```diff
@@ -10,31 +10,27 @@
 import SwiftUI
 
 class Email : ObservableObject{
-    @Published var email_addr: String
-    @Published var inbox: [InboxModel]
-    var requestUrl: URLComponents
+    @Published var email_addr: String?
+    @Published var inbox: [InboxModel]?
     
     //  init method
     init() {
-        self.email_addr = ""
-        self.inbox = [InboxModel]()
-        requestUrl = URLComponents()
-        requestUrl.scheme = "https"
-        requestUrl.host = "1secmail.com"
-        requestUrl.path = "/api/v1/"
+        reset()
+    }
+    
+    //resets the email
+    func reset() -> Void {
+        self.inbox = nil
+        self.email_addr = setNewEmailAddr()
         loadInbox()
```

Comment:
loadInbox() only needs to be called once, if its in reset, it'll be called multiple times. Move it to the init()